### PR TITLE
Improve configurable BBE with new structure

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
$subject
We need a more detailed BBEs for configurable feature.
Fixes https://github.com/ballerina-platform/ballerina-dev-website/issues/6019

## Approach
new structure under `Configurability` section
- Configurable variables
- Configuring via TOML files
- Configuring via CLI arguments

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: examples/configuring-via-cli/configuring_via_cli.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Configuring via command line arguments
+
+The values of the configurable variables can be configured through the command line arguments when executing the Ballerina program. The provided value is expected to be the `toString()` representation of the intended value.
+
+The command-line-based configuration is only supported for configurable variables of types `int`, `byte`, `float`, `boolean`, `string`, `decimal`, `enum`, and `xml`.
+The syntax `-Ckey=value` can be used to provide values through the command line parameters.
+
+For more information, see [Configure via command line arguments](/learn/configure-ballerina-programs/provide-values-to-configurable-variables/#provide-via-command-line-arguments/).
```
Comment (praneesha) [inline examples/configuring-via-cli/configuring_via_cli.md:8]:
```suggestion
For more information, see [Configure via command-line arguments](/learn/configure-ballerina-programs/provide-values-to-configurable-variables/#provide-via-command-line-arguments/).
```
We need to update the word `command-line` in the learn page also.

File: examples/configuring-via-cli/configuring_via_cli.metatags [hunk 2]
```diff
@@ -0,0 +1,2 @@
+description: This BBE demonstrates how to configure configurable variables in Ballerina via command line arguments.
```
Comment (praneesha) [inline examples/configuring-via-cli/configuring_via_cli.metatags:1]:
```suggestion
description: This BBE demonstrates how to configure configurable variables in Ballerina via command-line arguments.
```

File: examples/configuring-via-toml/configuring-via-toml.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Configuring via TOML files
```
Comment (praneesha) [inline examples/configuring-via-toml/configuring-via-toml.md:1]:
```suggestion
# Configure via TOML files
```

File: examples/configuring-via-toml/configuring_via_toml.bal [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+type UserTable table<UserInfo> key(username);
+
+enum HttpVersion {
+   HTTP_1_0 = "1.0",
+   HTTP_1_1 = "1.1",
+   HTTP_2_0 = "2.0"
+}
+
+// `configurable` variables of `float`, `string[]`, enum, `record`, and `table` types are initialized.
```
Comment (praneesha) [inline examples/configuring-via-toml/configuring_via_toml.bal:16]:
```suggestion
// The configurable variables of `float`, `string[]`, enum, `record`, and `table` types are initialized.
```

File: examples/configuring-via-toml/configuring_via_toml.metatags [hunk 5]
```diff
@@ -0,0 +1,2 @@
+description: This BBE demonstrates how to configure configurable variables in Ballerina via TOML file.
```
Comment (praneesha) [inline examples/configuring-via-toml/configuring_via_toml.metatags:1]:
```suggestion
description: This BBE demonstrates how to configure configurable variables in Ballerina via a TOML file.
```

File: examples/configuring-via-cli/configuring_via_cli.md (hunk 1 above)
Comment (HindujaB) [reply examples/configuring-via-cli/configuring_via_cli.md]: Created an issue for this https://github.com/ballerina-platform/ballerina-dev-website/issues/6194. I'll address in a separate PR.

**Review by praneesha — APPROVED** [review]

**Pull request merged**